staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller_test.go

```diff
@@ -0,0 +1,275 @@
+			},
+		},
+		"multiple-resourceslices-existing": {
+			testNodeUID: testNodeUID,
+			initialObjects: []runtime.Object{
+				newResourceSlice(testResourceSlice1, testNode, testDriveName, testPoolName, 1),
+				newResourceSlice(testResourceSlice2, testNode, testDriveName, testPoolName, 1),
+				newResourceSlice(testResourceSlice3, testNode, testDriveName, testPoolName, 1),
+			},
+			poolName: testPoolName,
+			inputDriverResources: &DriverResources{
+				Pools: map[string]Pool{
+					testPoolName: {
+						Devices: []resourceapi.Device{
+							{
+								Name: testDeviceName,
+							},
+						},
+					},
+				},
+			},
+			expectedResourceSlices: []resourceapi.ResourceSlice{
+				*newExpectResourceSlice(testResourceSlice1, testNode, string(testNodeUID), testDriveName, testPoolName, testDeviceName, false, 1),
+			},
+		},
+		"no-resourceslices-existing": {
+			testNodeUID:    testNodeUID,
+			initialObjects: []runtime.Object{},
+			poolName:       testPoolName,
+			inputDriverResources: &DriverResources{
+				Pools: map[string]Pool{
+					testPoolName: {
+						Devices: []resourceapi.Device{
+							{
+								Name: testDeviceName,
+							},
+						},
+					},
+				},
+			},
+			expectedResourceSlices: []resourceapi.ResourceSlice{
+				*newExpectResourceSlice(testNode+"-"+testDriveName+"-", testNode, string(testNodeUID), testDriveName, testPoolName, testDeviceName, true, 0),
+			},
+		},
+		"single-resourceslice-existing-with-different-resource-pool-generation": {
+			testNodeUID: testNodeUID,
+			initialObjects: []runtime.Object{
+				newResourceSlice(testResourceSlice1, testNode, testDriveName, testPoolName, 2),
+				newResourceSlice(testResourceSlice2, testNode, testDriveName, testPoolName, 1),
+				newResourceSlice(testResourceSlice3, testNode, testDriveName, testPoolName, 1),
+			},
+			poolName: testPoolName,
+			inputDriverResources: &DriverResources{
+				Pools: map[string]Pool{
+					testPoolName: {
+						Generation: 3,
+						Devices: []resourceapi.Device{
+							{
+								Name: testDeviceName,
+							},
+						},
+					},
+				},
+			},
+			expectedResourceSlices: []resourceapi.ResourceSlice{
+				*newExpectResourceSlice(testResourceSlice1, testNode, string(testNodeUID), testDriveName, testPoolName, testDeviceName, false, 3),
+			},
+		},
+		"single-resourceslice-existing-with-mismatching-resource-pool-name": {
+			testNodeUID: testNodeUID,
+			initialObjects: []runtime.Object{
+				newResourceSlice(testResourceSlice1, testNode, testDriveName, testPoolName, 1),
+			},
+			poolName: testPoolName,
+			inputDriverResources: &DriverResources{
+				Pools: map[string]Pool{
+					testPoolName1: {
+						Generation: 1,
+						Devices: []resourceapi.Device{
+							{
+								Name: testDeviceName,
+							},
+						},
+					},
+				},
+			},
+			expectedResourceSlices: nil,
+		},
+	}
+
+	for name, test := range testCases {
+		t.Run(name, func(t *testing.T) {
+			_, ctx := ktesting.NewTestContext(t)
+			// Create kubeClient
```

Comment:
nit: Do we really need this comment? The same goes for the rest of the test code. If the code is clear enough on its own, I don't think we need comments. But of course, if there's something tricky, some context or a strong opinion behind it, then it’s definitely worth including.